### ISSUE TITLE
Silent Install doesn't install Start Menu Shortcuts

### ISSUE THREAD

**Issue by pyoungberg** [NickeManarin/ScreenToGif#549]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%240%20Funded-%2300A156.svg" />](https://issuehunt.io/r/NickeManarin/ScreenToGif/issues/549)

We pushed the ScreenToGif.2.19.3.Setup.msi installer via Intune to our machines with the /silent switch and it installs, but no shortcut is placed in any of the users' Start Menus. Note, it's installing in the Per-User context so it should be running the installer as the currently signed in user. "C:\Program Files (x86)\ScreenToGif" exists on every machine we push it to, so we know it's installing. Can you have the msi installer drop a shortcut or a folder with a shortcut to "C:\ProgramData\Microsoft\Windows\Start Menu\Programs", even when the installer is run silently? Thank you!

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/NickeManarin/ScreenToGif/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

**Comment by NickeManarin** [comment]
Hi, the switches to add shortcuts are these:

> msiexec /i "ScreenToGif 2.19.3 Setup.msi" /quiet /norestart INSTALLDESKTOPSHORTCUT=yes INSTALLSHORTCUT=yes

https://github.com/NickeManarin/ScreenToGif/wiki/Tips-for-installing-this-app-using-the-command-line

I'm not sure if it works with Intune.

**Comment by pyoungberg** [comment]
Thanks, but this only works when manually executing ".\ScreenToGif.2.19.3.Setup.msi /quiet INSTALLSHORTCUT=yes". The properties (INSTALLSHORTCUT) are being ignored though when it's pushed through Intune. Can you confirm that they're marked as SecureCustomProperties, like shown in the example below?

https://stackoverflow.com/questions/39225934/customaction-and-msi-properties-ignored-when-deploying-through-intune

**Comment by NickeManarin** [comment]
Ah, it was false by default.
I can't test it, since I don't work with Intune. Could you test it?

```
Properties = new[]
{
    new Property("INSTALLDESKTOPSHORTCUT", "no") { Secure = true },
    new Property("INSTALLSHORTCUT", "no") { Secure = true }
},
```

[ScreenToGif 2.19.3 Setup.zip](https://github.com/NickeManarin/ScreenToGif/files/3826880/ScreenToGif.2.19.3.Setup.zip)

**Comment by pyoungberg** [comment]
Worked like a charm, thanks @NickeManarin. Successfully deployed to several machines through Intune.

**Comment by NickeManarin** [comment]
@pyoungberg Great! This change will be available with the next version of the app.

I see that this issue was funded on IssueHunt. 
Since the installer is not open sourced in here, there's no way to get the bounty. :/

**Comment by pyoungberg** [comment]
I canceled it on IssueHunt - You should see the payment come through PayPal in the next few days.

**Comment by NickeManarin** [comment]
@pyoungberg Oh, thank you! :D

**Comment by Asinin3** [comment]
Hi @pyoungberg , I'm running into this same issue on 2.39.
Running the installer. Using`msiexec /i "ScreenToGif 2.19.3 Setup.msi" /quiet /norestart INSTALLDESKTOPSHORTCUT=yes INSTALLSHORTCUT=yes` doesn't create any shortcuts. Running without the shortcut parameters doesn't create any in start menu or desktop either. Was something changed?

**Comment by DW-42** [comment]
I am able to reproduce this issue even with the latest version of ScreenToGif  (version 2.41.0)  when installed as the built-in Windows user 'NT AUTHORITY\SYSTEM'. The logs suggest that the installer is using the wrong values for the `ProgramMenuFolder` and `DesktopFolder` properties:

- `ProgramMenuFolder = C:\WINDOWS\system32\config\systemprofile\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\`
- `DesktopFolder = C:\WINDOWS\system32\config\systemprofile\Desktop\`

These values should be:

- `DesktopFolder = C:\Users\Public\Desktop\`
- `ProgramMenuFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\`

In summary, the MSI installer is likely hardcoding or incorrectly retrieving the 'NT AUTHORITY\SYSTEM' user profile paths instead of the expected public folders, resulting in the shortcuts not being created in the correct locations.
